airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/standard/PostgresStandardStateManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.postgres.standard;
+
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.source.postgres.internal.models.InternalModels.StateType;
+import io.airbyte.integrations.source.postgres.internal.models.StandardStatus;
+import io.airbyte.integrations.source.relationaldb.CursorInfo;
+import io.airbyte.integrations.source.relationaldb.models.DbState;
+import io.airbyte.integrations.source.relationaldb.state.StreamStateManager;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.AirbyteStreamState;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This state manager extends the StreamStateManager to enable writing the state_type and version
+ * keys to the stream state when they're going through the iterator Once we have verified that
+ * expanding StreamStateManager itself to include this functionality, this class will be removed
+ */
+public class PostgresStandardStateManager extends StreamStateManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(StreamStateManager.class);
+
+  private final List<AirbyteStateMessage> rawAirbyteStateMessages;
```

Comment:
unused parameter

Reply:
Ack will remove. Regarding updating the parent level stream state to have the keys that we need, we should do this in a separate PR as we can now roll this out as a pre-release. This change can be part of the larger refactor. cc: @akashkulk @subodh1810

Reply:
Let's create a tracking issue for this.